source/PluginDev/Assets/GooglePlayGames/Platforms/Native/NativeClient.cs

```diff
@@ -285,6 +286,56 @@ private static void LaunchBridgeIntent(IntPtr bridgedIntent)
                 AndroidJNIHelper.DeleteJNIArgArray(objectArray, jArgs);
             }
         }
+        
+        private AndroidJavaObject GetApiClient(GameServices services) {
+            return JavaUtils.JavaObjectFromPointer(C.InternalHooks_GetApiClient(services.AsHandle()));
+        }
+        
+        private string RetrieveUserEmail() {
+            string email;
+            using (AndroidJavaClass jc_plus = new AndroidJavaClass("com.google.android.gms.plus.Plus")) {
+                using (AndroidJavaObject jo_plusAccountApi = jc_plus.GetStatic<AndroidJavaObject>("AccountApi")) {
+                    Debug.Log("jo_plusAccountApi: " + (jo_plusAccountApi == null ? "NULL" : jo_plusAccountApi.ToString()));
+                    using (var apiClient = GetApiClient(mServices)) {
+                        Debug.Log("apiClient: " + (apiClient == null ? "NULL" : apiClient.ToString()));
+                        email  = jo_plusAccountApi.Call<string>("getAccountName", apiClient);
+                        Logger.d("Player email: " + email);
+                    }
+                }
+            }
+            return email;
+        }
+        
+        public string GetToken() {
+			if( String.IsNullOrEmpty(GameInfo.AndroidClientId) || ("__ANDROID_CLIENTID__" == GameInfo.AndroidClientId) )
+			{
+				throw new Exception("Client ID has not been set, cannot request access token.");
+			}
+            string token = null;
+            Debug.Log("Before RetrieveUserEmail");
+            string email = RetrieveUserEmail() ?? "NULL";
+            Debug.Log("After RetrieveUserEmail email: " + email);
+            string scope = "audience:server:client_id:" + GameInfo.AndroidClientId;
```

Comment:
If possible, I'd love for you guys to own that side of things and offer those options in a way you want to users.  I put this in based on @claywilkinson's feedback in the thread stating this looked great and asked for a pull request.

Potentially pull to a separate branch, then make the changes you want there.  I, like so many others in the thread, needed to get access to a access token that could be used to authenticate server-side and this 100% serves my purposes, and I'm assuming many of the other guys too.  This was more to "plug the gap" of not having this functionality at all, if you now want to take it and make it much, much better, I'd be delighted with that.  But I don't necessarily have the bandwidth to make it 100% bulletproof and exactly how Google want it to be.  I think that probably should be Google's responsibility.